new-lamassu-admin/src/pages/Services/schemas/helper.js

```diff
@@ -21,4 +24,35 @@ const leadingZerosTest = (value, context) => {
   return true
 }
 
-export { secretTest, leadingZerosTest }
+const buildCurrencyOptions = markets => {
+  return R.map(it => {
+    const unavailableCryptos = R.difference(ALL_CRYPTOS, markets[it])
+    const unavailableCryptosFiltered = R.difference(unavailableCryptos, [it]) // As the markets can have stablecoins to trade against other crypto, filter them out, as there can't be pairs such as USDT/USDT

```

Comment:
Since `unavailableCryptos` isn't used by itself you could replace these two lines with a single one. Something like this:

```js
const unavailableCryptos = R.difference(ALL_CRYPTOS, R.prepend(it, markets[it]))
```